pkg/metrics/server_test.go

```diff
@@ -13,19 +14,29 @@ import (
 	"github.com/prometheus/common/expfmt"
 )
 
-func runMetricsServer(t *testing.T) chan<- struct{} {
+var (
+	portOffset  int = 0
+	offsetMutex sync.Mutex
+)
+
+func runMetricsServer(t *testing.T) (int, chan<- struct{}) {
+	offsetMutex.Lock()
+	portOffset += 1
+	port := MetricsPort + portOffset
+	offsetMutex.Unlock()
```

Comment:
We always unlock here as well, but earlier, as soon as we have a calculated port for sure and we are confident that it won't interfere with another routine's port calculation (offset counter increase and port calculation being atomic operation thanks to the mutex).
I can `defer` of course, but it won't make any difference in this case, might only make some routine wait for longer on the mutex `Lock` operation.
However, since it is tests, we don't care much about that time difference.
So, if you think it will be better to `defer` anyways, I'll change it.

Reply:
@adambkaplan actually i've come up with a better idea, just using `sync/atomic` here. Atomic increment function does the cleaner job.